app/views/deploys/new.html.erb

```diff
@@ -12,67 +12,65 @@
-            <div class="arrow" style="top: 20px"></div>
+      <% end %>
+      </div>
+      <div class="col-lg-6">
+        <div id="ref-problem-warning" class="col-lg-10 popover show right alert alert-warning hidden" style="max-width: 100%">
+          <div class="arrow" style="top: 20px"></div>
 
-            <h3 class="popover-title">Problems detected:
-              <button type="button" class="close" data-dismiss="alert" aria-hidden="true">×</button>
-            </h3>
-            <div class="popover-content">
-              <ul id="ref-problem-list"></ul>
-            </div>
+          <h3 class="popover-title">Problems detected:
+            <button type="button" class="close" data-dismiss="alert" aria-hidden="true">×</button>
+          </h3>
+          <div class="popover-content">
+            <ul id="ref-problem-list"></ul>
           </div>
         </div>
       </div>
+    </div>
 
-      <% if @stage.allow_redeploy_previous_when_failed? %>
-        <% form.object.redeploy_previous_when_failed = true %>
-        <%= form.input :redeploy_previous_when_failed, as: :check_box %>
-      <% end %>
+    <% if @stage.allow_redeploy_previous_when_failed? %>
+      <% form.object.redeploy_previous_when_failed = true %>
+      <%= form.input :redeploy_previous_when_failed, as: :check_box %>
+    <% end %>
 
-      <%= render 'recent_releases' %>
-      <%= render 'currently_deployed' %>
+    <%= render 'recent_releases' %>
+    <%= render 'currently_deployed' %>
 
-      <%= Samson::Hooks.render_views(:deploy_form, self, project: @project, form: form) %>
+    <%= Samson::Hooks.render_views(:deploy_form, self, project: @project, form: form) %>
 
-      <div class="form-group" id="new-deploy-buttons">
-        <div class="col-lg-offset-2 col-lg-10">
-          <%= form.submit "", class: "btn btn-primary", disabled: Lock.locked_for?(@stage, current_user), data: {disable_with: false} %>
-          <%= link_to "Cancel", :back, class: 'btn btn-default' %>
-        </div>
+    <div class="form-group" id="new-deploy-buttons">
+      <div class="col-lg-offset-2 col-lg-10">
+        <%= form.submit "", class: "btn btn-primary", disabled: Lock.locked_for?(@stage, current_user), data: {disable_with: false} %>
+        <%= link_to "Cancel", :back, class: 'btn btn-default' %>
       </div>
+    </div>
 
-      <div id="deploy-confirmation">
-        <ul class="nav nav-tabs">
-          <%= render 'changeset/tab_list' %>
-        </ul>
-        <div class="tab-content deploy-details">
-          <div class="tab-pane changeset-placeholder">Loading...</div>
-        </div>
+    <div id="deploy-confirmation">
+      <ul class="nav nav-tabs">
+        <%= render 'changeset/tab_list' %>
+      </ul>
+      <div class="tab-content deploy-details">
+        <div class="tab-pane changeset-placeholder">Loading...</div>
       </div>
-    </fieldset>
+    </div>
```

Comment:
should have been easy if there were comments like <!-- End of deploy-confirmation div --> hard to check if the syntax is correct in this way. Replacing fieldset with div.

Reply:
github has a "no whitespace" option for diffs